producer-c/producer-cloudwatch-integ/CanaryStreamCallbacks.cpp

```diff
@@ -0,0 +1,216 @@
+/**
+ * Kinesis Video Producer Continuous Retry Stream Callbacks
+ */
+#define LOG_CLASS "CanaryStreamCallbacks"
+#include "CanaryStreamCallbacks.h"
+
+STATUS createCanaryStreamCallbacks(Aws::CloudWatch::CloudWatchClient* cwClient,
+                                   PCHAR pStreamName,
+                                   PStreamCallbacks* ppStreamCallbacks, PCanaryStreamCallbacks pCanaryStreamCallbacks)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    Aws::CloudWatch::Model::Dimension dimension;
+
+    CHK(ppStreamCallbacks != NULL, STATUS_NULL_ARG);
+
+    // Allocate the entire structure
+    CHK(pCanaryStreamCallbacks != NULL, STATUS_NOT_ENOUGH_MEMORY);
+
+    // Set the version, self
+    pCanaryStreamCallbacks->streamCallbacks.version = STREAM_CALLBACKS_CURRENT_VERSION;
+    pCanaryStreamCallbacks->streamCallbacks.customData = (UINT64) pCanaryStreamCallbacks;
+    pCanaryStreamCallbacks->timeOfNextKeyFrame = new std::map<UINT64, UINT64>();
+    pCanaryStreamCallbacks->pStreamName = pStreamName;
```

Comment:
Careful with this one - you are only storing a pointer here. Make sure the actual stream name is available throughout the object lifetime or make a local copy into the struct. Not yet sure why you need this - will be looking in